packages/core/contracts/cross-chain-oracle/README.md

```diff
@@ -129,3 +131,65 @@ To finalize **Arbitrum** transactions see the docs [here](https://github.com/Off
 To finalize **Optimism** transactions see the equivalent script [here](https://github.com/ethereum-optimism/optimism/blob/develop/packages/message-relayer/src/exec/withdraw.ts). Optimism's Etherscan also provides a UI, similar to Arbitrum, which can be found [here](https://optimistic.etherscan.io/txsExit).
 
 **Boba** at present auto-finalizes all L2->L1 transactions without any required user intervention.
+
+## Step-by-step guide to deployments for Non-rollup chains
+
+The steps below explain how to deploy the cross-chain oracle onto non-rollup chains using the `Admin_ChildMessenger`. This enables a multisig to act as the "messenger" thereby enabling the UMA Optimistic oracle to be deployed onto chains that do not have a supporting canonical bridge.
+
+1. Start by exporting some environment variables (or storing them in a .env):
+
+```sh
+# Set only the following environment variables based on which L2 you're deploying to. For example, if you're deploying to xDAI, set NODE_URL_100.
+# When running against a forked network, set the URL to http://localhost:<PORT>
+export NODE_URL_100=<XDAI_URL>
+export MNEMONIC="Your 12-word mnemonic here"
+```
+
+2. Deploy l2 Contracts
+
+```sh
+yarn hardhat deploy --network [xdai] --tags l2-admin-xchain,Registry
+```
+
+Add the deployed [xdai] Registry, finder, OracleSpoke, GovernorSpoke and Admin_ChildMessenger contracts to the associated networks file.
+
+3. Verify contracts:
+
+```sh
+# xDai (etherscan-verify does not work on xdai so we use hardhat verify)
+yarn hardhat --network [xdai] verify <EACH-DEPLOYED ADDRESS IN STEP 2> <ASSOCIATED CONSTRUCTOR PARAMS FROM 2>
+```
+
+3. Setup l2 contracts
+
+```sh
+yarn hardhat setup-l2-xchain --network [xdai]
+```
+
+4. At this point, the cross chain contract suite setup is complete. We will now deploy the `OptimisticOracle` and required contracts to the L2.
+
+```sh
+yarn hardhat deploy --network [xdai] --tags OptimisticOracle,IdentifierWhitelist,AddressWhitelist,Store
+```
+
+8. Verify contracts:
+
+```sh
+# xDai (etherscan-verify does not work on xdai so we use hardhat verify)
+yarn hardhat --network xdai verify <EACH-DEPLOYED ADDRESS IN STEP 4> <ASSOCIATED CONSTRUCTOR PARAMS FROM 4>
```

Comment:
Does `yarn hardhat sourcify --network xdai` work? This is suggested here: https://docs.blockscout.com/for-users/smart-contract-interaction/verifying-a-smart-contract/contracts-verification-via-sourcify.

I only ask because, like `etherscan-verify`, sourcify is much less painful than the plain verify call.